### ISSUE TITLE
fine tune confusion

### ISSUE THREAD

**Issue by zhly0** [davidsandberg/facenet#235]
I use the model 20170216-091149 to fine tune,I have a train sample ,class num is 6000,I want to fine tune the last layer logits,so I restore all the variable except logits,and send the layer 'logits' to the facenet.train's paramater 'update_gradient_vars' to let it just update the last logits layer,but I get confused that,I have only update the last layer logits layer,then the layer before logits remain uchanged,so the embedding i get shoud not change  to the same data.and my validate on lfw should be the same to the model 20170216-091149 ,which is  98.2%  ,but only 74% when fine tuning the last layer logits after some epoch.is there something wrong in my procedure?

**Comment by davidsandberg** [comment]
Hi,
Before I experimented with fine tuning using triplet loss, but I did it a little bit different than you. In my setup the network looks exactly the same as when training using softmax, except that triplet loss on the embeddings is the optimization critera. Also, the whole network is being updated already from the start of the training.

`pyhton src/train_tripletloss.py --data_dir ~/datasets/casia/casia_maxpy_mtcnnalign_182_160 --image_size 160 --model_def models.inception_resnet_v1 --lfw_dir ~/datasets/lfw/lfw_mtcnnalign_160 --weight_decay 1e-4 --optimizer RMSPROP --learning_rate -1 --keep_probability 0.8 --random_crop --random_flip --learning_rate_schedule_file data/learning_rate_schedule_tripletloss_casia.txt --pretrained_model /media/data/DeepLearning/models/facenet/20170214-092102/model-20170214-092102.ckpt-80000 --people_per_batch 720 --images_per_person 5`

I haven't managed to improve the performance significantly over training using softmax but at least the performance doesn't crash.
It would be nice to hear if you can improve on the classifier training.

**Comment by zhly0** [comment]
@davidsandberg  thanks for reply,I stopped try to fine tune the network,but try some small network to train from the beginning,and now I get a lfw accuracy of 95.8% on a small network with a model size less than 50M.thanks for your framework!

**Comment by davidsandberg** [comment]
Nice!!

**Comment by speculaas** [comment]
Dear All,
About fine-tuning, is there any particular strategy?

Do you update every trainable variables?
My rough idea currently is, maybe just fine-tuning some higher level weights would be good enough?
Maybe like tensorflow-for-poets?

https://codelabs.developers.google.com/codelabs/tensorflow-for-poets/#0

My reasoning is that the first few layers of David's pretrained model is probably already good at recognizing basic visual elements such as oriented edges and opposing colors etc.

Maybe we only need to fine tune upper layers' weight?

Reason of my asking is because I used --pretrained_model option to update your pre-trained model (model-20180402-114759.ckpt-275, vggface2) using only 60 persons, and the result performance totally crashed. 

My test set contained 99 people and using my newly retrained model, they all grouped into a SINGLE cluster after dbscan.

Any suggestion and experience of different ways of fine-tuning?

**Comment by sarthakahuja11** [comment]
@speculaas  I agree with your reasoning of fine tuning only upper layers' weights. I am working in a similar scenario. What did you do to finally get better results?

**Comment by speculaas** [comment]
Hi Sarthakahuja, 
Actually, I probably only have disappointing results. 
My aim is to enhance asian people's distinguishing power.
I used some Asian data set to fine tune bottleneck, logits, block8
![tensorboard facenet screenshot from 2018-04-17 14-24-22](https://user-images.githubusercontent.com/31201176/41777982-bf8c2a6a-765f-11e8-86ae-97006e15c9eb.png)

The result is my lfw validation accuracy dropped. 

Maybe fine tune some more layers?
Maybe generate an Asian pair.txt to see if Asian recognition power improved after fine tuning?
** https://github.com/davidsandberg/facenet/issues/376
Maybe try learning without forgetting method?

BR,
Jimmy

**Comment by Vis5** [comment]
@speculaas I'm trying to work on the same thing. Can you share us your Asian face dataset?

**Comment by speculaas** [comment]
Dear Vis, 

Here is an Asian face dataset : https://github.com/deepinsight/insightface/issues/256

BR,
Jimmy

**Comment by hungnv21292** [comment]
hi @speculaas 
Could you please share with me how to generate an Asian pair.txt to evaluate my finetune model.

Thank you so much.

**Comment by speculaas** [comment]
hi viethungtsdv, 
sorry I haven't done it.
I only found out some generous guy provided instructions as I stated above : #256 
I didn't get a chance to actually go through the steps described in #256
